gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala

```diff
@@ -80,60 +77,80 @@ case class FakeRowAdaptor(child: SparkPlan) extends UnaryExecNode with GlutenPla
     copy(child = newChild)
 }
 
-object GlutenColumnarRules {
-
-  // TODO: support Insert clause
-
+object GlutenWriterColumnarRules {
   // TODO: support ctas in Spark3.4, see https://github.com/apache/spark/pull/39220
   // TODO: support dynamic partition and bucket write
   //  1. pull out `Empty2Null` and required ordering to `WriteFilesExec`, see Spark3.4 `V1Writes`
   //  2. support detect partition value, partition path, bucket value, bucket path at native side,
   //     see `BaseDynamicPartitionDataWriter`
-  def isGlutenInsertInto(cmd: DataWritingCommand): Boolean = {
+  def isNativeParquetAppliable(cmd: DataWritingCommand): Boolean = {
+
+    if (!GlutenConfig.getConf.enableNativeParquetWriter) {
+      return false
+    }
+
     cmd match {
       case command: CreateDataSourceTableAsSelectCommand =>
         if (command.table.provider.contains("velox")) {
           throw new UnsupportedOperationException(
             "Velox file format does not support create table as select.")
         }
-        false
+        "parquet".equals(command.table.provider.get)
       case command: InsertIntoHadoopFsRelationCommand
-          if command.fileFormat.isInstanceOf[GlutenParquetFileFormat] =>
-        if (command.partitionColumns.nonEmpty || command.bucketSpec.nonEmpty) {
+          if command.fileFormat.isInstanceOf[ParquetFileFormat] =>
+        if (
+          GlutenConfig.isCurrentBackendVelox
+          && (command.partitionColumns.nonEmpty || command.bucketSpec.nonEmpty)
+        ) {
           throw new UnsupportedOperationException(
             "Velox file format does not support dynamic partition write and bucket write.")
         }
         true
       case command: InsertIntoHiveDirCommand =>
-        command.storage.outputFormat.get
-          .equals("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat")
-      case _ => false
+        "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat".equals(
+          command.storage.outputFormat.get)
+      case command: InsertIntoHiveTable =>
+        "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat".equals(
+          command.table.storage.outputFormat.get)
+      case _: CreateHiveTableAsSelectCommand =>
+        throw new UnsupportedOperationException(
+          "native writer cannot recognize command: " + cmd.getClass.getName +
+            " please append `using parquet` in your CTAS query")
+      case _ =>
+        throw new UnsupportedOperationException(
+          "native writer cannot recognize command: " + cmd.getClass.getName)
```

Comment:
Here, we are only doing an `isNativeParquetAppliable` analysis on the plan. For inapplicable situations, it should return false rather than throwing an exception.